app/stac_api/models.py

```diff
@@ -236,6 +237,26 @@ def update_extent(self, item_properties_datetime):
             logger.error('Updating the collection extent due to item update failed: %s', err)
             raise ValidationError(_("Updating the collection extent due to item update failed."))
 
+    def update_spatial_bbox(self, geometry):
+        '''
+        updates the collection's spatial extent when an item is updated.
+        :param geometry: the geometry of the item
+        This function generates a new extent regarding all the items with the same
+        collection foreign key. If there is no spatial bbox yet, the one of the geometry of the
+        item is being used.
+        Warning: This function may be slow, when there are millions of items (because of the query)
+        '''
+        if self.extent["spatial"]["bbox"]:
+            if self.extent["spatial"]["bbox"] is None:
+                self.extent["spatial"]["bbox"] = GEOSGeometry(geometry).extent
+            else:
+                qs = Item.objects.filter(collection_id=self.pk)
+                union_geometry = GEOSGeometry('Multipolygon EMPTY')
+                for item in qs:
+                    union_geometry = union_geometry.union(item.geometry)
+                self.extent["spatial"]["bbox"] = union_geometry.extent
```

Comment:
looping over all items in the collection whenever an item is saved will work, but will not be performant for more than a few houndred items (remember that we'll have collections with a few hundred-thousand items). I suggest to compare here just the extent new `geometry` with the existing extent on the collection and adapt if necessary.
I would keep this functionality in a dedicated mgmt command (`recalculate_collection_cache` or the like), where we can explicitly trigger the recalculation of these values from all items if necessary.

Reply:
From my understanding of the bbox the update is only needed in these cases:

1. New item
    a. new item bbox is outside of collection bbox => no loop needed for update
2. Item update
    a. old item inside collection bbox without neighbor border
        i. new item outside collection bbox => no loop needed for update
    b. old item inside collection bbox with neighbor border
        i. new item outside collection bbox => no loop needed for update
        ii. new item inside collection bbox and neighbor border changed => loop needed for update

To resume only the 4 use case needs an update of collection and only one needs a loop:
* 1.a => no loop needed
* 2.a.i => no loop needed
* 2.b.i => no loop needed
* 2.b.ii => loop needed

The question is how often an item geometry update would happen ? would it happen at all ? And then is the item.save() method blocking API request or admin page ?
When the item update geometry doesn't happen often and the save() method is not blocking, then I think the loop only when needed is fine. When it is blocking and we can make it asynchrone then it is also fine to have the loop. When it might be often updated we might want to search for a better loop algorithm. When the checks above are too complex and the item geometry update is a rare case then the loop asynchrone might be an acceptable solution in my opinion.

Reply:
@ltshb thanks for that, I lost a day (because I did not listen to your advice and tried an other soloution)